src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -282,6 +282,12 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetTotalOperationalHours(uint32_t & total
 
     totalOperationalHours = static_cast<uint32_t>(totalHours + deltaTime < UINT32_MAX ? totalHours + deltaTime : UINT32_MAX);
 
+    if (deltaTime > 0)
```

Comment:
I know that you have only modified the existing code, but I have some observation that maybe we could try to address.

Generally, I don't like that this method `GetTotalOperationalHours` is almost 1:1 duplicated by the `UpdateOperationalHours`. I think it would be good to create a separate method that contains the common part and use it in both places. It could e.g. return two values - total operational hours and delta from the previous store. This way we could call here UpdateOperationalHours in delta > 0  and remove `SetSavedOperationalHoursSinceBoot` at all.

Reply:
sure, I've done some refactoring. now `UpdateOperationalHours` i a common method which can be called everytime we try to get totaloperationalhours, and every `CONFIG_CHIP_OPERATIONAL_TIME_SAVE_INTERVAL` hours